documentation/ha.markdown

```diff
@@ -0,0 +1,275 @@
+
+PuppetDB HA has two parts: first, Puppet Server is configured to send commands
+and queries to multiple PuppetDB servers. Second, those servers are configured
+to periodically reconcile their differences, transferring any records which they
+are missing. This process is pull-based and runs on a configurable interval.
+
+PuppetDB replication is in principle a multi-master system. You can issue
+commands to any server, and they can be reconciled with any other server without
+conflicts. But, in order to minimize confusion when using exported resources, we
+recommend that only node be conventionally treated as primary.
+[See below](#exportedResources) for further discussion.
+
+PuppetDB HA is available in Puppet Enterprise 2016.5 and later.
+
+Manual Installation and Configuration
+----
+
+1. Provision two PuppetDB nodes. Designate one of these as your _primary_
+   PuppetDB, and the other as the _replica_. Be sure that each PuppetDB is using
+   its own PostgreSQL cluster, since HA doesn't buy you anything if you have
+   single database. In this guide, we will give them the hostnames
+   `primary-puppetdb.mycorp.net` and `replica-puppetdb.mycorp.net`.
+
+2. Configure each PuppetDB to pull from the other. This guide will use
+   HOCON-style configuration. See the
+   [PE Configuration Documentation][pe-configure] for further details.
+
+  a. Configure primary-puppetdb to pull from replica-puppetdb by placing this in its
+     config file:
+
+         [sync]
+         server_urls = https://replica-puppetdb.mycorp.net:8081
+         intervals = 2m
+
+  b. Configure replica-puppetdb to pull from primary-puppetdb:
+
+         [sync]
+         server_urls = https://primary-puppetdb.mycorp.net:8081
+         intervals = 2m
+
+3. Configure the Puppet master to point to both of your PuppetDB instances. In
+   your puppetdb.conf, use this as your server configuration (in the 'main' section):
+
+        server_urls = https://primary-puppetdb.mycorp.net:8081,https://replica-puppetdb.mycorp.net:8081
+        sticky_read_failover = true
+        command_broadcast = true
+
+   It is important that your primary PuppetDB appear first in this list, so it
+   will always be tried first when submitting new data from the Puppet master.
+
+4. Restart your PuppetDBs and your Puppet master.
+
+Deployment tips
+----
+
+- Each PuppetDB server should be configured to use a separate PostgreSQL
+  instance.
+
+- If you are using multiple compile masters, be sure that their clocks are in
+  sync (using NTP, for example). PuppetDB relies on timestamps for discerning
+  which data is newer for any given node, so the clocks on your compile masters
+  need to be within `runinterval` of each other.
+
+- The timeout for HTTP connections made by Puppet Server, including those to
+  PuppetDB, can be configured using the
+  `http-client.connect-timeout-milliseconds` key. This defaults to 120 seconds.
+  The correct value will depend on your infrastructure, but 10 seconds is a good
```

Comment:
seems weird to state that we default to 120 and then turn around and recommend 10